utils/obsproc/NetCDFToIodaConverter.h

```diff
@@ -58,57 +65,108 @@ namespace gdasapp {
-
-      // Update the group with the location dimension
-      ioda::NewDimensionScales_t
-        newDims {ioda::NewDimensionScale<int>("Location", iodaVars.location)};
-      ioda::ObsGroup ogrp = ioda::ObsGroup::generate(group, newDims);
-
-      // Set up the float creation parameters
-      ioda::VariableCreationParameters float_params;
-      float_params.chunk = true;               // allow chunking
-      float_params.compressWithGZIP();         // compress using gzip
-      float missing_value = util::missingValue(missing_value);
-      float_params.setFillValue<float>(missing_value);
-
-      // Create the IODA variables
-      ioda::Variable adtIodaDatetime =
-        ogrp.vars.createWithScales<float>("Metadata/dateTime",
-                                          {ogrp.vars["Location"]}, float_params);
-      // TODO(Mindo): Get the date info from the netcdf file
-      adtIodaDatetime.atts.add<std::string>("units", {"seconds since 9999-04-15T12:00:00Z"}, {1});
-
-      ioda::Variable adtIodaObsVal =
-        ogrp.vars.createWithScales<float>("ObsValue/"+variable_,
-                                          {ogrp.vars["Location"]}, float_params);
-      ioda::Variable adtIodaObsErr =
-        ogrp.vars.createWithScales<float>("ObsError/"+variable_,
-                                          {ogrp.vars["Location"]}, float_params);
-
-      // Write adt obs value to group
-      adtIodaObsVal.writeWithEigenRegular(iodaVars.obsVal);
-
-      // Write adt obs error to group
-      adtIodaObsErr.writeWithEigenRegular(iodaVars.obsError);
+      int myrank  = comm.rank();
+      int nobs(0);
+      oops::Log::debug() << "ooooooooooooo my rank : " << myrank << comm.size() << std::endl;
+      if (myrank <= inputFilenames_.size()-1) {
+        providerToIodaVars(inputFilenames_[myrank], iodaVars);
+        nobs = iodaVars.location;
+        oops::Log::debug() << "--- iodaVars.location: " << iodaVars.location << std::endl;
+        oops::Log::debug() << "--- iodaVars.obsVal: " << iodaVars.obsVal << std::endl;
+      }
+
+      // Get the total number of obs across pe's
+      comm.allReduce(nobs, nobs, eckit::mpi::sum());
+      oops::Log::debug() << " my rank : " << myrank
+                         << " Num pe's: " << comm.size()
+                         << " nobs: " << nobs << std::endl;
+      gdasapp::IodaVars iodaVarsAll(nobs);
+
+      // Gather obsVal's
+      gatherObs(comm, iodaVars.obsVal, iodaVarsAll.obsVal);
+      gatherObs(comm, iodaVars.obsError, iodaVarsAll.obsError);
+      gatherObs(comm, iodaVars.preQc, iodaVarsAll.preQc);
+
+      oops::Log::debug() << "--- all nobs: " << iodaVarsAll.obsVal.size() << std::endl;
+      oops::Log::debug() << "--- all obsVal: " << iodaVarsAll.obsVal << std::endl;
+      oops::Log::debug() << "--- all obsError: " << iodaVarsAll.obsError << std::endl;
+      oops::Log::debug() << "--- all preQc: " << iodaVarsAll.preQc << std::endl;
```

Comment:
Could this code go inside the `if` statement. Do you really need to gather in all the MPI tasks or just the one that writes the file?

Reply:
We're going to introduce a thinning step before the writing to ioda. This is probably not the most memory efficient way to do this, but I'm planning to have iodaVarAll available from all pe's ... this might change though ...